src/storage/src/hummock/store/local_hummock_storage.rs

```diff
@@ -229,17 +260,7 @@ impl LocalStateStore for LocalHummockStorage {
         key_range: TableKeyRange,
         read_options: ReadOptions,
     ) -> impl Future<Output = StorageResult<Self::IterStream<'_>>> + Send + '_ {
```

Comment:
Can we change this to `async` as well?

Reply:
Actually there is a more general question: is it possible to change the trait in store.rs to async trait

Reply:
> Can we change this to async as well?

Yes we can, but it will unnecessarily wrap a extra layer of async/await.

> is it possible to change the trait in store.rs to async trait

Currently we cannot. async fn in trait is actually a sugar for defining future return type.
For example, 
```
trait Trait {
    async fn f(&self);
}
```
actually defines 
```
trait Trait {
    fn f(&self) -> impl Future<Output = ()> + '_;
}
```

Under this sugar, we are not able to add `Send` constraint in the trait definition. Therefore in the trait we can only write as `impl Future<Output = ()> + Send + '_` to verbosely specify the `Send` constraint.

Reply:
Makes sense. Thanks for the explanation.